app/Jobs/CreateRockTheVotePostInRogue.php

```diff
@@ -0,0 +1,392 @@
+use Chompy\Traits\ImportToRogue;
+use Illuminate\Queue\InteractsWithQueue;
+use Illuminate\Contracts\Queue\ShouldQueue;
+use Illuminate\Foundation\Bus\Dispatchable;
+
+class CreateRockTheVotePostInRogue implements ShouldQueue
+{
+    use Dispatchable, InteractsWithQueue, Queueable, ImportToRogue;
+
+    /**
+     * The record to be created into a post from the csv.
+     *
+     * @var array
+     */
+    protected $record;
+
+    /**
+     * Create a new job instance.
+     *
+     * @return void
+     */
+    public function __construct($record)
+    {
+    	$this->record = $record;
+    }
+
+    /**
+     * Execute the job to create a Turbo Vote post in Rogue.
+     *
+     * @return array
+     */
+    public function handle(Rogue $rogue)
+    {
+    	$shouldProcess = $this->scrubRecord($this->record);
+
+        if ($shouldProcess) {
+            info('progress_log: Processing: ' . $this->record['Email address']);
+
+            $referralCode = $this->record['Tracking Source'];
+            $referralCodeValues = $this->parseReferralCode($referralCode);
+
+            try {
+                $user = $this->getOrCreateUser($this->record, $referralCodeValues);
+            } catch (\Exception $e) {
+                info('There was an error with that user: ' . $this->record['Email address'], [
+                    'Error' => $e->getMessage(),
+                ]);
+            }
+
+            if (isset($user)) {
+                $post = $rogue->getPost([
+                    'campaign_id' => (int) $referralCodeValues['campaign_id'],
+                    'northstar_id' => $user->id,
+                    'type' => 'voter-reg',
+                ]);
+
+                if (! $post['data']) {
+                    $rtvCreatedAtMonth = strtolower(Carbon::parse($this->record['Started registration'])->format('F-Y'));
+                    $sourceDetails = isset($referralCodeValues['source_details']) ? $referralCodeValues['source_details'] : null;
+                    $postDetails = $this->extractDetails($this->record);
+
+                    $postData = [
+                        'campaign_id' => (int) $referralCodeValues['campaign_id'],
+                        'campaign_run_id' => (int) $referralCodeValues['campaign_run_id'],
+                        'northstar_id' => $user->id,
+                        'type' => 'voter-reg',
+                        'action' => $rtvCreatedAtMonth . '-rockthevote',
+                        'status' => $this->translateStatus($this->record['Status'], $this->record['Finish with State']),
+                        'source' => 'rock-the-vote',
+                        'source_details' => $sourceDetails,
+                        'details' => $postDetails,
+                    ];
+
+                    try {
+                        $post = $rogue->createPost($postData);
+
+                        if ($post['data']) {
+                            info('post created in rogue for ' . $this->record['id']);
```

Comment:
there's no "id" column in RTV csvs, I believe previously we were using email address?

Reply:
ah great catch! Updated :)